core/store/src/db.rs

```diff
@@ -222,6 +256,14 @@ impl Database for TestDB {
         Box::new(iterator)
     }
 
+    fn iter_prefix<'a>(
+        &'a self,
+        col: DBCol,
+        key_prefix: &'a [u8],
+    ) -> Box<dyn Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a> {
```

Comment:
```suggestion
    ) -> impl Iterator<Item = (Box<[u8]>, Box<[u8]>)> + 'a {
```
Does this work?

Reply:
`impl Trait` is not supported in traits.